### PR TITLE
Add peoples_speech

### PR DESCRIPTION
**Pull request by yfyeung**
The People's Speech Dataset is among the world’s largest English speech recognition corpus today that is licensed for academic and commercial usage under CC-BY-SA and CC-BY 4.0. It includes 30,000+ hours of transcribed speech in English languages with a diverse set of speakers. This open dataset is large enough to train speech-to-text systems and crucially is available with a permissive license. Just as ImageNet catalyzed machine learning for vision, the People’s Speech will unleash innovation in speech research and products that are available to users across the globe.

- $dl_dir/peoples_speech
     This directory contains the following files downloaded from
      https://huggingface.co/datasets/MLCommons/peoples_speech
 
   - test
   - train
   - validation

| Part    | Tarfile Number | Unzipped Size | 
| -- | -- | -- |
| dirty          | 9900 | 1.4T  |
| dirty_sa    | 1635  | 137G |
| clean        | 4481  | 375G |
| clean_sa  | 767    | 64G |
| validation | 24     | 2.3G |
| test           | 42     | 4.1G |

| Stage | Time | Comment | 
| -- | -- | -- | 
| Stage 0: Download data | - | Use wget to download the dataset manuallly |
| Stage 1: Prepare  People's Speech manifest | 3 hours | num_jobs=32 |
| Stage 2: Prepare musan manifest | Very little | - |
| Stage 3: Preprocess People's Speech manifest | 35 minutes | - |
| Stage 4: Compute fbank for valid and test subsets of People's Speech | 4 minutes | 1 GPU,  total size 3G|
| Stage 5: Split train subset into pieces | 20 minutes  | - |
| Stage 6: Compute features for train subset of People's Speech | 8 hours  | 4 processes with 1 GPUs, total size 925G |
| Stage 7: Compute fbank for musan | 15 minutes  | - |
| Stage 8: Prepare BPE based lang | 38 minutes  | - |

### Subset Describe
- dirty
```shell
╒═══════════════════════════╤═════════════╕
│ Cuts count:               │ 5476898     │
├───────────────────────────┼─────────────┤
│ Total duration (hh:mm:ss) │ 21698:39:42 │
├───────────────────────────┼─────────────┤
│ mean                      │ 14.3        │
├───────────────────────────┼─────────────┤
│ std                       │ 1.9         │
├───────────────────────────┼─────────────┤
│ min                       │ 1.0         │
├───────────────────────────┼─────────────┤
│ 25%                       │ 14.5        │
├───────────────────────────┼─────────────┤
│ 50%                       │ 14.8        │
├───────────────────────────┼─────────────┤
│ 75%                       │ 14.9        │
├───────────────────────────┼─────────────┤
│ 99%                       │ 15.0        │
├───────────────────────────┼─────────────┤
│ 99.5%                     │ 15.0        │
├───────────────────────────┼─────────────┤
│ 99.9%                     │ 15.0        │
├───────────────────────────┼─────────────┤
│ max                       │ 16.7        │
├───────────────────────────┼─────────────┤
│ Recordings available:     │ 5476898     │
├───────────────────────────┼─────────────┤
│ Features available:       │ 0           │
├───────────────────────────┼─────────────┤
│ Supervisions available:   │ 5476898     │
╘═══════════════════════════╧═════════════╛
SUPERVISION custom fields:
Speech duration statistics:
╒══════════════════════════════╤═════════════╤══════════════════════╕
│ Total speech duration        │ 21698:39:42 │ 100.00% of recording │
├──────────────────────────────┼─────────────┼──────────────────────┤
│ Total speaking time duration │ 21698:39:42 │ 100.00% of recording │
├──────────────────────────────┼─────────────┼──────────────────────┤
│ Total silence duration       │ 00:00:00    │ 0.00% of recording   │
╘══════════════════════════════╧═════════════╧══════════════════════╛
```

- dirty_sa
```shell
╒═══════════════════════════╤════════════╕
│ Cuts count:               │ 548014     │
├───────────────────────────┼────────────┤
│ Total duration (hh:mm:ss) │ 2130:12:07 │
├───────────────────────────┼────────────┤
│ mean                      │ 14.0       │
├───────────────────────────┼────────────┤
│ std                       │ 2.5        │
├───────────────────────────┼────────────┤
│ min                       │ 1.0        │
├───────────────────────────┼────────────┤
│ 25%                       │ 14.5       │
├───────────────────────────┼────────────┤
│ 50%                       │ 14.8       │
├───────────────────────────┼────────────┤
│ 75%                       │ 14.9       │
├───────────────────────────┼────────────┤
│ 99%                       │ 15.0       │
├───────────────────────────┼────────────┤
│ 99.5%                     │ 15.0       │
├───────────────────────────┼────────────┤
│ 99.9%                     │ 15.0       │
├───────────────────────────┼────────────┤
│ max                       │ 15.1       │
├───────────────────────────┼────────────┤
│ Recordings available:     │ 548014     │
├───────────────────────────┼────────────┤
│ Features available:       │ 0          │
├───────────────────────────┼────────────┤
│ Supervisions available:   │ 548014     │
╘═══════════════════════════╧════════════╛
SUPERVISION custom fields:
Speech duration statistics:
╒══════════════════════════════╤════════════╤══════════════════════╕
│ Total speech duration        │ 2130:12:07 │ 100.00% of recording │
├──────────────────────────────┼────────────┼──────────────────────┤
│ Total speaking time duration │ 2130:12:07 │ 100.00% of recording │
├──────────────────────────────┼────────────┼──────────────────────┤
│ Total silence duration       │ 00:00:00   │ 0.00% of recording   │
╘══════════════════════════════╧════════════╧══════════════════════╛
```

- clean
```shell
╒═══════════════════════════╤════════════╕
│ Cuts count:               │ 1501271    │
├───────────────────────────┼────────────┤
│ Total duration (hh:mm:ss) │ 5894:44:45 │
├───────────────────────────┼────────────┤
│ mean                      │ 14.1       │
├───────────────────────────┼────────────┤
│ std                       │ 2.4        │
├───────────────────────────┼────────────┤
│ min                       │ 1.0        │
├───────────────────────────┼────────────┤
│ 25%                       │ 14.5       │
├───────────────────────────┼────────────┤
│ 50%                       │ 14.8       │
├───────────────────────────┼────────────┤
│ 75%                       │ 14.9       │
├───────────────────────────┼────────────┤
│ 99%                       │ 15.0       │
├───────────────────────────┼────────────┤
│ 99.5%                     │ 15.0       │
├───────────────────────────┼────────────┤
│ 99.9%                     │ 15.0       │
├───────────────────────────┼────────────┤
│ max                       │ 16.6       │
├───────────────────────────┼────────────┤
│ Recordings available:     │ 1501271    │
├───────────────────────────┼────────────┤
│ Features available:       │ 0          │
├───────────────────────────┼────────────┤
│ Supervisions available:   │ 1501271    │
╘═══════════════════════════╧════════════╛
SUPERVISION custom fields:
Speech duration statistics:
╒══════════════════════════════╤════════════╤══════════════════════╕
│ Total speech duration        │ 5894:44:45 │ 100.00% of recording │
├──────────────────────────────┼────────────┼──────────────────────┤
│ Total speaking time duration │ 5894:44:45 │ 100.00% of recording │
├──────────────────────────────┼────────────┼──────────────────────┤
│ Total silence duration       │ 00:00:00   │ 0.00% of recording   │
╘══════════════════════════════╧════════════╧══════════════════════╛
```

- clean_sa
```shell
╒═══════════════════════════╤═══════════╕
│ Cuts count:               │ 257093    │
├───────────────────────────┼───────────┤
│ Total duration (hh:mm:ss) │ 991:43:29 │
├───────────────────────────┼───────────┤
│ mean                      │ 13.9      │
├───────────────────────────┼───────────┤
│ std                       │ 2.8       │
├───────────────────────────┼───────────┤
│ min                       │ 1.0       │
├───────────────────────────┼───────────┤
│ 25%                       │ 14.5      │
├───────────────────────────┼───────────┤
│ 50%                       │ 14.8      │
├───────────────────────────┼───────────┤
│ 75%                       │ 14.9      │
├───────────────────────────┼───────────┤
│ 99%                       │ 15.0      │
├───────────────────────────┼───────────┤
│ 99.5%                     │ 15.0      │
├───────────────────────────┼───────────┤
│ 99.9%                     │ 15.0      │
├───────────────────────────┼───────────┤
│ max                       │ 16.2      │
├───────────────────────────┼───────────┤
│ Recordings available:     │ 257093    │
├───────────────────────────┼───────────┤
│ Features available:       │ 0         │
├───────────────────────────┼───────────┤
│ Supervisions available:   │ 257093    │
╘═══════════════════════════╧═══════════╛
SUPERVISION custom fields:
Speech duration statistics:
╒══════════════════════════════╤═══════════╤══════════════════════╕
│ Total speech duration        │ 991:43:29 │ 100.00% of recording │
├──────────────────────────────┼───────────┼──────────────────────┤
│ Total speaking time duration │ 991:43:29 │ 100.00% of recording │
├──────────────────────────────┼───────────┼──────────────────────┤
│ Total silence duration       │ 00:00:00  │ 0.00% of recording   │
╘══════════════════════════════╧═══════════╧══════════════════════╛
```

### CODE REVIEW

File: egs/peoples_speech/ASR/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  # to data/musan
+  mkdir -p data/manifests
+  if [ ! -e data/manifests/.musan.done ]; then
+    lhotse prepare musan $dl_dir/musan data/manifests
+    touch data/manifests/.musan.done
+  fi
+fi
+
+if [ $stage -le 3 ] && [ $stop_stage -ge 3 ]; then
+  log "Stage 3: Preprocess People's Speech manifest"
```
Comment (csukuangfj) [inline egs/peoples_speech/ASR/prepare.sh:105]: Please change the comment for this stage. It is the same as stage 1 at present.

File: egs/peoples_speech/ASR/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  if [ ! -e data/fbank/.preprocess_complete ]; then
```
Comment (csukuangfj) [inline egs/peoples_speech/ASR/prepare.sh:106]: data/fbank does not exist before stage 4.

**Review by csukuangfj — CHANGES_REQUESTED** [review]

File: egs/peoples_speech/ASR/prepare.sh (hunk 1 above)
Comment (yfyeung) [reply egs/peoples_speech/ASR/prepare.sh:105]: Stage 1 is prepare, and Stage 3 is preprocess

File: egs/peoples_speech/ASR/prepare.sh (hunk 2 above)
Comment (yfyeung) [reply egs/peoples_speech/ASR/prepare.sh:107]: øk, fixed it.

File: egs/peoples_speech/ASR/prepare.sh [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        find "data/fbank/peoples_speech_cuts_dirty_raw.jsonl.gz"
+	find "data/fbank/peoples_speech_cuts_dirty_sa_raw.jsonl.gz"
+	find "data/fbank/peoples_speech_cuts_clean_raw.jsonl.gz"
+	find "data/fbank/peoples_speech_cuts_clean_sa_raw.jsonl.gz"
+      )
+      gunzip -c ${file} | awk -F '"' '{print $30}' > $lang_dir/transcript_words.txt
+
+      # Ensure space only appears once
+      sed -i 's/\t/ /g' $lang_dir/transcript_words.txt
+      sed -i 's/[ ][ ]*/ /g' $lang_dir/transcript_words.txt
```
Comment (csukuangfj) [inline egs/peoples_speech/ASR/prepare.sh:185]:
```suggestion
      sed -i 's/ +/ /g' $lang_dir/transcript_words.txt
```
is simpler.

Comment (yfyeung) [reply egs/peoples_speech/ASR/prepare.sh]: OK

File: egs/peoples_speech/ASR/local/preprocess_peoples_speech.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            continue
+
+        logging.info(f"Normalizing text in {partition}")
+        for sup in m["supervisions"]:
+            text = str(sup.text)
+            orig_text = text
+            sup.text = normalize_text(sup.text)
+            text = str(sup.text)
+            if len(orig_text) != len(text):
+                logging.info(
```
Comment (csukuangfj) [inline egs/peoples_speech/ASR/local/preprocess_peoples_speech.py:94]: Will it print lots of logs? If so, can we limit number of lines to print?

Comment (yfyeung) [reply egs/peoples_speech/ASR/local/preprocess_peoples_speech.py:95]: OK

File: egs/peoples_speech/ASR/local/compute_fbank_peoples_speech_valid_test.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This file computes fbank features of the CommonVoice dataset.
```
Comment (csukuangfj) [inline egs/peoples_speech/ASR/local/compute_fbank_peoples_speech_valid_test.py:20]:
```suggestion
This file computes fbank features of the People's Speech dataset.
```

**Review by csukuangfj — APPROVED** [review]
Thanks!

**Comment by chenguoguo** [comment]
Is there a RESULTS file for this dataset? Have you evaluated it internally? @yfyeung @csukuangfj

**Comment by yfyeung** [comment]
Currently, there is no result for Peoples' Speech dataset. We are training a non-streaming and a streaming Zipformer on multi-dataset (LibriSpeech + GigaSpeech + CommonVoice + Peoples' Speech).

**Pull request merged**